_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+
+    +   Contains both tutorials and problems of basic Algorithms and Data Structures.
+
+    +   Tutorials are easy to understand and covers most of the topic.
+
+    +   Can be easily completed within a month or two.
+
+    +   A **'must do'** for beginners.
+
++   **[A2OJ](https://www.a2oj.com) :**
+
+    +   We are often unable to find good problems of a particular topic, that's where a2oj comes to rescue.
+
+    +   Problems are from online judges like SPOJ, UVa, Codeforces and are classified according to topic and difficulty.
+
+## Contests:
+
++   **[Codechef](https://www.codechef.com) :**
+
+    +   Organizes 3 rated contests every month, namely, Long Challenge(10 days), LunchTime(3 hrs) and Cook-Off(2.5 hrs).
+
+    +   Long Challenge is best for learning new concepts and improving known ones. Do participate in Long Challenge every month and do read editorials of problems not solved during contest and try to solve them.
+
+    +   Short contests are best for improving speed and accuracy.
+
++   **[Codeforces](http://codeforces.com/) :**
+
+    +   4-5 Shorts contests(2 hrs) every month.
+
+    +   Very effective in improving speed and accuracy.
+
+    +   Less time, more problems and more participants makes it tougher and more interesting than Codechef.
+    
+---
+
+# Getting Started
+
+In order to solve a problem, you first read the problem, code the solution and submit the solution online. A verdict is shown from which you can see how your code performed against the test cases ran by the online judge.
+
+## Problem Statement:
+
+Statement starts with few paragraphs explaining the problem. Read it thoroughly until you understand what is to be solved.
+
+After that, there's *Input and output* format explaining how input and output is formatted. NEVER output anything else than what is asked. For example, do not print stuff like "Enter a number: ".
+
+Then, *Constraints* are given which will give you the idea of how big the input is and whether your approach will work or not. 
+Constraints include --
++   **Time limit :** Given in *seconds*. Time less than which your code should calculate and print the output.
+
++   **Memory limit :** Given in *MB*. Maximum amount of memory your code should use.
+
++   **Input Size :** Maximum possible input size, ie, input bigger than this won't be there in test cases.
+
+Then, there will be *Sample Input* and *Sample Output* to give you an idea of I/O formats and to give you a better understanding of the problem.
+
+Lastly, there will be *Explanation*(might be absent) which explains how sample output is calculated.
+
+## Solving the problem:
+
+After reading and understanding the problem, a solution is to be coded for it. A given problem can have many possible solutions but you have to find that solution which comes under the constraints. In order to see if your code fulfills the time and memory limits, you need to do an *Order Analysis* of your solution.
+
+**Order Analysis :** 
```

Comment:
Google searching order analysis leads to unrelated stuff: https://www.google.co.in/search?q=Order+Analysis